src/main.rs

```diff
@@ -66,26 +67,41 @@ struct SourceMapJson<'a> {
   names: &'a Vec<String>,
 }
 
+#[derive(Debug)]
+struct InputFile {
+  filename: String,
+  source: String
+}
+
 pub fn main() -> Result<(), std::io::Error> {
   let cli_args = CliArgs::parse();
   let project_root = std::env::current_dir()?;
+  
+  let mut files: Vec<InputFile> = Vec::new();
 
-  // If we're given an input file, read from it and adjust its name.
-  //
-  // If we're not given an input file and stdin was redirected, read
-  // from it and create a fake name. Return an error if stdin was not
-  // redirected (otherwise the program will hang waiting for input).
-  //
-  let (filename, source) = match &cli_args.input_file {
-    Some(f) => {
-      let absolute_path = fs::canonicalize(f)?;
-      let filename = pathdiff::diff_paths(absolute_path, &project_root).unwrap();
-      let filename = filename.to_string_lossy().into_owned();
-      let contents = fs::read_to_string(f)?;
-      (filename, contents)
-    }
+  match &cli_args.input_file {
+    Some(g) => {
+      let globs = glob(g);
+
+      match glob(g) {
+        Ok(file_paths) => {
+          for file in file_paths {
+            if let Ok(f) = file {
```

Comment:
I was also wondering if you could give me some pointers on this code, there's quite a bit of nested pattern matching going on, is there a cleaner/more idiomatic way to approach this?

Reply:
One question I have is if we need to expand globs ourselves here or not? Usually the shell (e.g. bash) will expand globs before executing the program, so the glob itself will never reach it, it'll just be a list of files.

Reply:
During development I used the following command to run the cli:
```
⚡ cargo run --features="cli" "*.css"
    Finished dev [unoptimized + debuginfo] target(s) in 0.58s
     Running `target/debug/lightningcss '*.css'`
html {
  color: red;
}

body {
  background: green;
}
```

Having the quotes around the glob does seem needed, if you provide the glob without quotes, it does indeed get expanded by the shell, and result in the following:

```
⚡ cargo run --features="cli" *.css
    Finished dev [unoptimized + debuginfo] target(s) in 0.42s
     Running `target/debug/lightningcss test.css test2.css`
error: Found argument 'test2.css' which wasn't expected, or isn't valid in this context

USAGE:
    lightningcss [OPTIONS] [INPUT_FILE]

For more information try --help
```